### PR TITLE
upgrade to yarn 4

### PR DESCRIPTION
**Pull request by mifi**
Note: breaking change (requires node>=18)

TODO
- [ ] replace np with something that works with yarn 4 (and update readme) #169

### CODE REVIEW

**Comment by mifi** [comment]
I've tried to tinker with upgrading np to support yarn berry and it seems to be working, but I haven't done tests yet: https://github.com/mifi/np/pull/1

**Comment by kvz** [comment]
upstream np accepts yarn 4 now as indicated in https://github.com/mifi/np/pull/1, so we could likely upgrade np along with yarn and settle here

**Comment by mifi** [comment]
yes but I think it would be a breaking change?

**Comment by remcohaszing** [comment]
I’m running into the same problem in #176. Vitest also requires Node.js >=18. Although I personally prefer to test against all supported Node.js versions, I suggest we drop tests for Node.js 14 and 16 so we can upgrade Yarn and Vitest. The upcoming TypeScript migration will define a target matching Node.js 14, so the odds of breaking Node.js 14 support are slim.

**Comment by mifi** [comment]
yea im fine with not testing against old versions for now

**Review by remcohaszing — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -11,10 +11,9 @@
     "mp3"
   ],
   "author": "Tim Koschuetzki <tim@transloadit.com>",
-  "packageManager": "yarn@1.22.19+sha1.4ba7fc5c6e704fce2066ecbfb0b0d8976fe62447",
+  "packageManager": "yarn@4.0.1",
   "engines": {
-    "node": ">= 14.16",
-    "yarn": ">= 1.22.19"
+    "node": ">= 18"
```
Comment (remcohaszing) [inline package.json:16]: We should revert this. The `engines` field tells the engines users should use. So this should _never_ contain any package managers, and the `node` engine should be restored to its original value.

```suggestion
    "node": ">= 14.16",
```

**Review by remcohaszing — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (mifi) [reply package.json:16]: you mean revert the node change but not revert the yarn change?

Comment (remcohaszing) [reply package.json:16]: Yes

**Review by remcohaszing — APPROVED** [review]

**Review by kvz — APPROVED** [review]

**Pull request merged**